### DISCUSSION TITLE
Reloading all routes after an action is erasing params

### DISCUSSION THREAD

**Discussion by EmberRandall** [remix-run/remix#7242] [Q&A] [answered]
Hi Remix community! I'm fairly new to Remix, and running into an issue that I'm not sure how to solve. 

The app we're building is displaying a set of tickets, which can exist in folders (like in Gmail). We have a couple of nested routes -- `app/tickets`, with child `app/tickets/$folderId` -- that display the folders and the tickets themselves, respectively. When a user drags tickets into a new folder, we're currently using `useSubmit` to call the `action` method defined on `app/tickets`, and that moves the tickets into the folder.

The problem is, when that `action` gets called, all the routes are reloaded. When initially dragging the tickets, the user is on a page like `app/tickets/123`, but, when all the routes are reloaded, the parent `app/tickets` route reloads too, and the `folderId` param in the url is disappearing. So the user is kicked back to `app/tickets`, and is no longer viewing the folder that they wanted to be viewing. How do I prevent that?

**Comment by sergiodxa** [comment, marked as answer]
The `useSubmit` hook, like the `Form` component, cause a navigation to the `action` URL. You can use `useFetcher().submit` to do the request without causing a navigation.

**Reply by EmberRandall** [reply]
That definitely fixes the navigation issue, thank you! But it seems like `useActionData` is then never called? The `action` itself is called, but I have a `useEffect` listening for changes in the action data because I need to get the result from the submission, and that's no longer being called

**Reply by EmberRandall** [reply]
Never mind, figured that one out, the fetcher itself has the data!
